tensorflow/lite/micro/micro_interpreter.cc

```diff
@@ -86,7 +86,11 @@ void MicroInterpreter::Init(MicroProfiler* profiler) {
   context_.ReportError = MicroContextReportOpError;
   context_.GetTensor = MicroContextGetTensor;
   context_.GetEvalTensor = MicroContextGetEvalTensor;
+#ifndef TF_LITE_DISABLE_PROFILING
   context_.profiler = profiler;
+#else
+  context_.profiler = nullptr;
+#endif
```

Comment:
Why not simply pass a nullptr when we create a `MicroInterpreter` ?

Reply:
then i'd have to find all places where they are created and make sure that i don't pass in a profiler, whenever `TF_LITE_DISABLE_PROFILING`. that seems to me a bad interface, to shift the responsibility to the caller.

we could also do it in the places where `Init()` is called, but then there would be repetition.

i think it's cleaner the way i did it, but maybe i mis-understood your point?

Reply:
I was thinking from the point of view of minimalizing the number of tfmicro changes, but you have a good argument.